pkg/core/mempool/mem_pool.go

```diff
@@ -154,13 +160,25 @@ func (mp *Pool) Add(t *transaction.Transaction, fee Feer) error {
 		mp.lock.Unlock()
 		return ErrDup
 	}
-	err := mp.checkTxConflicts(t, fee)
+	err := mp.checkAttributesConflicts(t, fee)
+	if err != nil {
+		mp.lock.Unlock()
+		return err
+	}
+	err = mp.checkTxConflicts(t, fee)
 	if err != nil {
 		mp.lock.Unlock()
 		return err
```

Comment:
And if we're to get here the side-effect from `checkAttributesConflicts` will remain even though the transaction was rejected. This must be fixed.

Reply:
Same thing of course with any other return from this function except the successful one.

Reply:
`checkTxConflicts` is the only case when we should pay attention to `return`. The second `return` (mempool overflow) is not the case because if there's at least one conflicting transaction, we should remove it first and only after that we should add the new transaction.

However, adding conflicting hashes to the mp.conflicts list should be done after overflow chek.